src/main/java/net/rubygrapefruit/platform/internal/jni/AbstractFileEventFunctions.java

```diff
@@ -44,6 +45,84 @@ public AbstractWatcherBuilder(BlockingQueue<FileWatchEvent> eventQueue) {
          * @see FileWatcher#startWatching(Collection)
          */
         public abstract FileWatcher start() throws InterruptedException;
+
+        /**
+         * Starts the file watcher with a thread consuming the events and
+         * feeding them to the given {@link FileWatchEvent.Handler}.
+         *
+         * The started thread will terminate once the returned
+         * {@link FileWatcher} is {@link FileWatcher#close() closed}.
```

Comment:
```suggestion
         * {@link FileWatcher} is successfully {@link FileWatcher#close() closed}.
```

When closing doesn't work for some reason, the thread won't terminate. I think the watcher should terminate the thread on close always, even when it can't close the file watcher.

Reply:
It should now terminate if the close is unsuccessful, i.e. it times out.